server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -846,6 +859,15 @@ public final CommitStats commitStats() {
      */
     public abstract long getPersistedLocalCheckpoint();
 
+    /**
+     * @return the latest checkpoint that has been processed but not necessarily persisted.
+     * Also see {@link #getPersistedLocalCheckpoint()}
+     */
+    public long getProcessedLocalCheckpoint() {
+        // default implementation
+        return 0L;
```

Comment:
Sorry didn't catch this before, I'm not sure we want a default implementation here, this will impact NRTReplicationEngine & ReadOnlyEngine and return a misleading value.

Reply:
Fixed